src/azure-cli-core/azure/cli/core/commands/parameters.py

```diff
@@ -60,19 +60,32 @@ def completer(prefix, action, parsed_args, **kwargs):#pylint: disable=unused-arg
             return [r.name for r in get_resources_in_subscription(resource_type=resource_type)]
     return completer
 
-def get_enum_type_completion_list(enum_type=None):
-    ENUM_MAPPING_KEY = '_value2member_map_'
-    def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-argument
-        return list(enum_type.__dict__[ENUM_MAPPING_KEY].keys())
-    return completer
-
 def get_generic_completion_list(generic_list):
     def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-argument
         return generic_list
     return completer
 
-def get_enum_choices(enum_type):
-    return [x.value for x in enum_type]
+class CaseInsenstiveList(list): # pylint: disable=too-few-public-methods
+    def __contains__(self, other):
+        other_lower = other.lower()
+        for val in self:
+            if other_lower == val.lower():
+                return True
+        return False
+
+def enum_choice_list(enum_type):
+    """ Creates the argparse choices and type kwargs for a supplied enum type. """
+    enum_type_values = [x.value for x in enum_type]
+    def _type(value):
+        for val in enum_type_values:
+            if val.lower() == value.lower():
+                return val
+        return value
+    params = {
+        'choices': CaseInsenstiveList(enum_type_values),
```

Comment:
Correct. VM size list still uses its "get_vm_sizes" method call. This is only used for enum types. Also, I did not update the choice lists in Storage (my favorite) to use this, because the enums used in storage are not SDK-style enums and they would not work.